src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -105,28 +106,63 @@ mutateRequestToQuery (Insert mainQi iCols body onConflct putConditions returning
     ])
   where
     cols = BS.intercalate ", " $ pgFmtIdent <$> S.toList iCols
-mutateRequestToQuery (Update mainQi uCols body logicForest returnings) =
-  if S.null uCols
+
+mutateRequestToQuery (Update mainQi uCols body logicForest (range, rangeId) returnings)
+  | S.null uCols =
     -- if there are no columns we cannot do UPDATE table SET {empty}, it'd be invalid syntax
     -- selecting an empty resultset from mainQi gives us the column names to prevent errors when using &select=
     -- the select has to be based on "returnings" to make computed overloaded functions not throw
-    then SQL.sql ("SELECT " <> emptyBodyReturnedColumns <> " FROM " <> fromQi mainQi <> " WHERE false")
-    else
-      "WITH " <> normalizedBody body <> " " <>
-      "UPDATE " <> SQL.sql (fromQi mainQi) <> " SET " <> SQL.sql cols <> " " <>
-      "FROM (SELECT * FROM json_populate_recordset (null::" <> SQL.sql (fromQi mainQi) <> " , " <> SQL.sql selectBody <> " )) _ " <>
-      (if null logicForest then mempty else "WHERE " <> intercalateSnippet " AND " (pgFmtLogicTree mainQi <$> logicForest)) <> " " <>
-      SQL.sql (returningF mainQi returnings)
+    SQL.sql $ "SELECT " <> emptyBodyReturnedColumns <> " FROM " <> fromQi mainQi <> " WHERE false"
+
+  | range == allRange =
+    "WITH " <> normalizedBody body <> " " <>
+    "UPDATE " <> mainTbl <> " SET " <> SQL.sql nonRangeCols <> " " <>
+    "FROM (SELECT * FROM json_populate_recordset (null::" <> mainTbl <> " , " <> SQL.sql selectBody <> " )) _ " <>
+    whereLogic <> " " <>
```

Comment:
Looks like this change always adds a `WHERE` clause to the `UPDATE` query - even if the request is completely unconditional.

This breaks using `pg-safeupdate`. The request now goes through instead of failing because of a previously unconditional `UPDATE` query.

Reply:
Note that `pg-safeupdate` is not required anymore because we're disallowing [unfiltered patch](https://github.com/PostgREST/postgrest/issues/1959#issuecomment-1152813181)/[unfiltered delete](https://github.com/PostgREST/postgrest/issues/2314) unless a `limit` is applied.

> Looks like this change always adds a WHERE clause to the UPDATE query - even if the request is completely unconditional.

Yeah, because of the above, if no filter is added then `WHERE false` is done.

One thing better with the `pg-safeupdate` safeguard is that it fails with an error while the `WHERE false` will not do anything.
That's only the case of unfiltered UPDATE though. For unfiltered DELETE, [we're going to fail with an error](https://github.com/PostgREST/postgrest/blob/47a599371fc65d89da782ad31ce003ea0088b863/test/spec/Feature/Query/DeleteSpec.hs#L304-L318).

(I need to go back to this one because it will be inconsistent otherwise)

Would we need to keep compatibility with `pg-safeupdate` now that we have this? Why?

---

Not sure but maybe we could do `LIMIT 0` instead of `WHERE false`.

Reply:
> Yeah, because of the above, if no filter is added then WHERE false is done.

Oh, hell. I thought this was a review on https://github.com/PostgREST/postgrest/pull/2311.

Reply:
Addressing this in https://github.com/PostgREST/postgrest/pull/2405/

Reply:
> Looks like this change always adds a WHERE clause to the UPDATE query - even if the request is completely unconditional.

I think the change in this PR still allows the full table update without conditionals. The one that disallowed this was #2311 if I'm not mistaken.

Reply:
Compare https://github.com/PostgREST/postgrest/pull/2418